celery/backends/base.py

```diff
@@ -491,7 +491,7 @@ def _get_result_meta(self, result,
                     if hasattr(request, 'delivery_info') and
                     request.delivery_info else None,
                 }
-                if getattr(request, 'stamps'):
+                if getattr(request, 'stamps', None):
```

Comment:
would you mind adding a unit test for this?

Reply:
Done :-)